### ISSUE TITLE
frexp operator frame tracing error with dynamic None

### ISSUE THREAD

**Issue by jthakurH** [pytorch/pytorch#120511]
### 🐛 Describe the bug

frexp operator with out variant causes frame tracing error when run with dynamic=None.
If we disable dynamic shape i.e. dynamic=False for torch.compile then it works fine
use below code to reproduce the error

### Error logs

```
Traceback (most recent call last):
  File "/tmp/sgn.py", line 25, in <module>
    res = compiled_model(params)
  File "/tmp/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1514, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/tmp/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1523, in _call_impl
    return forward_call(*args, **kwargs)
  File "/tmp/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 489, in _fn
    return fn(*args, **kwargs)
  File "/tmp/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1514, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/tmp/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1523, in _call_impl
    return forward_call(*args, **kwargs)
  File "/tmp/sgn.py", line 10, in forward
    result = self.op(**kwarg)
  File "/tmp/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 655, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state)
  File "/tmp/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 727, in _convert_frame
    result = inner_convert(frame, cache_entry, hooks, frame_state)
  File "/tmp/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 353, in _convert_frame_assert
    if not has_tensor_in_frame(frame):
  File "/tmp/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 1610, in inner_fn
    return fn(*args, **kwargs)
  File "/tmp/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 235, in has_tensor_in_frame
    if has_tensor(value):
  File "/tmp/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 224, in has_tensor
    seen_ids[obj_id] = any(has_tensor(getattr(obj, v)) for v in obj._fields)
AttributeError: 'torch.return_types.frexp_out' object has no attribute '_fields'. Did you mean: 'n_fields'?

```

### Minified repro

```
import torch


class NnWrapper(torch.nn.Module):
    def __init__(self, op):
        super().__init__()
        self.op = op

    def forward(self, kwarg):
        result = self.op(**kwarg)
        return result


model = NnWrapper(torch.frexp)
compiled_model = torch.compile(model)

shapes = [[1, 1, 2, 3], [1, 1, 3, 3]]
for shape in shapes:
    ifm = torch.randn(shape, dtype=torch.bfloat16)
    out = [
        torch.randn(shape, dtype=torch.bfloat16),
        torch.randint(low=-50, high=50, size=shape, dtype=torch.int32),
    ]
    params = {"input": ifm, "out": out}
    res = compiled_model(params)
    print(f"res: {res}")

```

### Versions

[pip3] numpy==1.26.4
[pip3] torch==2.2.0a0
[pip3] torchaudio==2.2.0
[pip3] torchdata==0.7.1
[pip3] torchmetrics==1.2.1
[pip3] torchtext==0.17.0
[pip3] torchvision==0.17.0

cc @ezyang @msaroufim @bdhirsh @anijain2305 @zou3519

**Comment by ezyang** [comment]
This is the same as https://github.com/pytorch/pytorch/issues/120482, fixed by https://github.com/pytorch/pytorch/pull/120567

**Comment by atalman** [comment]
I can confirm this is resolved in 2.3 nightly build:
```
res: torch.return_types.frexp_out(
mantissa=tensor([[[[-0.5977, -0.6523, -0.5859],
          [-0.5117, -0.9727, -0.6914]]]], dtype=torch.bfloat16),
exponent=tensor([[[[-7,  1, -4],
          [ 0, -1, -1]]]], dtype=torch.int32))
res: torch.return_types.frexp_out(
mantissa=tensor([[[[ 0.6680, -0.8984, -0.5195],
          [-0.9219,  0.5195, -0.7148],
          [ 0.8438,  0.6523,  0.5703]]]], dtype=torch.bfloat16),
exponent=tensor([[[[-1, -2,  1],
          [ 1, -1,  0],
          [ 0, -1,  0]]]], dtype=torch.int32))
```

Failing on 2.2.0 with the same error as in description

On 2.1.0 release I get different error:
```
File "/home/atalman/miniconda3/envs/py310/lib/python3.10/site-packages/torch/_inductor/codegen/triton.py", line 26, in <module>
    from ..triton_heuristics import AutotuneHint
  File "/home/atalman/miniconda3/envs/py310/lib/python3.10/site-packages/torch/_inductor/triton_heuristics.py", line 43, in <module>
    from triton.runtime.jit import get_cuda_stream, KernelInterface
ImportError: cannot import name 'get_cuda_stream' from 'triton.runtime.jit' (/home/atalman/miniconda3/envs/py310/lib/python3.10/site-packages/triton/runtime/jit.py)
```

**Comment by huydhn** [comment]
Note that this is only fixed in 2.3.0, it's still there on 2.2.2 final RC, but that's probably ok
